tests/queries/0_stateless/03285_constant_compare_in_and_chain.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TABLE test
```

Comment:
this test confuses me. do we really need it?

Reply:
It tests the first few lines [here](https://github.com/ClickHouse/ClickHouse/blob/5a23a5d8647c4d718f87ee2899839ba108b24111/src/Analyzer/Passes/LogicalExpressionOptimizerPass.cpp#L1055), because they are not algorithm related, so put it here?

Reply:
I think other tests should have covered this line?

For example, if change this line to 
```
   if (node->getNodeType() != QueryTreeNodeType::FUNCTION)
            throw Exception ...;
```

I am 99% sure that some test should fail.

So I bet we don't need this test

Reply:
I extracted the tests from previously failed run. Then let me remove it.